### PR TITLE
Gulp task that creates production build of the app

### PR DESCRIPTION
**Pull request by bryk**
This is the last dangling Gulp task. Next I'll unify them and wire up the build system.

### CODE REVIEW

File: build/build.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+ * Following steps are done here:
+ *  1. Vendor CSS and JS files are concatenated and minified.
+ *  2. index.html is minified.
+ *  3. CSS and JS assets are suffixed with version hash.
+ *  4. Everything is saved in the dist directory.
+ */
+gulp.task('build', ['index:prod', 'assets'], function () {
+  let htmlFilter = gulpFilter('*.html', { restore: true });
+  let vendorCssFilter = gulpFilter('**/vendor.css', { restore: true });
+  let vendorJsFilter = gulpFilter('**/vendor.js', { restore: true });
```
Comment (mgr01) [inline build/build.js]: Please remove spaces after { and before } in object literals.
(see: https://google.github.io/styleguide/javascriptguide.xml?showone=Code_formatting#Code_formatting)

File: build/build.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    .pipe(gulpRevReplace())
+    .pipe(htmlFilter)
+    .pipe(gulpMinifyHtml({
+      empty: true,
+      spare: true,
+      quotes: true
+    }))
+    .pipe(htmlFilter.restore)
+    .pipe(gulp.dest(conf.paths.dist))
+    .pipe(gulpSize({ showFiles: true }));
```
Comment (mgr01) [inline build/build.js]: Also here.

File: build/build.js (hunk 1 above)
Comment (bryk) [inline build/build.js]: Right, done. I haven't wired up ESlint yet, so that's why there are such inconsistencies.

File: build/build.js (hunk 2 above)
Comment (bryk) [inline build/build.js]: Done.

**Comment by bryk** [comment]
PTAL

**Pull request merged**